core/icons/mutator_icon.ts

```diff
@@ -0,0 +1,358 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as goog from '../../closure/goog/goog.js';
+goog.declareModuleId('Blockly.Mutator');
+
+import {Abstract} from '../events/events_abstract.js';
+import type {Block} from '../block.js';
+import {BlockChange} from '../events/events_block_change.js';
+import {BlocklyOptions} from '../blockly_options.js';
+import {BlockSvg} from '../block_svg.js';
+import type {Connection} from '../connection.js';
+import {Coordinate} from '../utils/coordinate.js';
+import * as dom from '../utils/dom.js';
+import * as eventUtils from '../events/utils.js';
+import {IHasBubble} from '../interfaces/i_has_bubble.js';
+import {Icon} from './icon.js';
+import {MiniWorkspaceBubble} from '../bubbles/mini_workspace_bubble.js';
+import {MUTATOR_TYPE} from './icon_types.js';
+import {Rect} from '../utils/rect.js';
+import {Size} from '../utils/size.js';
+import {Svg} from '../utils/svg.js';
+import type {WorkspaceSvg} from '../workspace_svg.js';
+
+export class MutatorIcon extends Icon implements IHasBubble {
+  /** The type string used to identify this icon. */
+  static readonly TYPE = MUTATOR_TYPE;
+
+  /**
+   * The weight this icon has relative to other icons. Icons with more positive
+   * weight values are rendered farther toward the end of the block.
+   */
+  static readonly WEIGHT = 1;
+
+  /** The size of this icon in workspace-scale units. */
+  private readonly SIZE = 17;
+
+  /** The distance between the root block and the workspace edges. */
+  private readonly MARGIN = 16;
+
+  /** The bubble used to show the mini workspace to the user. */
+  private miniWorkspaceBubble: MiniWorkspaceBubble | null = null;
+
+  /** The root block in the mini workspace. */
+  private rootBlock: BlockSvg | null = null;
+
+  /** The PID tracking updating the workkspace in response to user events. */
+  private updateWorkspacePid: ReturnType<typeof setTimeout> | null = null;
+
+  constructor(
+    private readonly flyoutBlockTypes: string[],
+    protected readonly sourceBlock: BlockSvg
+  ) {
+    super(sourceBlock);
+  }
+
+  getType() {
+    return MutatorIcon.TYPE;
+  }
+
+  initView(pointerdownListener: (e: PointerEvent) => void): void {
+    if (this.svgRoot) return; // Already initialized.
+
+    super.initView(pointerdownListener);
+
+    // Square with rounded corners.
+    dom.createSvgElement(
```

Comment:
My understanding was that loading from a separate .svg file means more http requests or developers need to bundle their local `media` directory with our assets (since I don't think we bundle our own .svg assets).

But I'm not sure! For now I plan on keeping this logic the same, and if we want to refactor it separately, I can file an issue to track doing so.